src/Api/Order.php

```diff
@@ -129,13 +129,24 @@ public static function place_order( \WP_REST_Request $request ) {
 		$order_id = $checkout->create_order();
 		$order = new \WC_Order( $order_id );
 
-		do_action( Hooks::AFTER_ORDER, $request, $cart );
-
 		// If the user is not logged in, we need to pass the billing and shipping address to the order.
 		if ( ! is_user_logged_in() ) {
 			$order = self::update_guest_order( $request, $order );
+
+			if ( is_wp_error( $order ) ) {
+				// Because we can have errors in $order, delete the order and return the error.
+				if ( in_array( get_post( $order_id )->post_type, wc_get_order_types(), true ) ) {
+					wp_delete_post( $order_id );
```

Comment:
Do you mean this ? @mitogh I check if the post_type is in order types using the `wc_get_order_types()` function, please let me know your thoughts on this, thank you!